### PR TITLE
⚡️(app-desk) clean build pages

### PR DESCRIPTION
**Pull request by AntoLC**
### Prob:
Next.js transpiles all the files present in the
`pages` directory. But we don't want to transpile
the providers neither the Layout components.

image

### Solution:
We export these components to their features.

Screenshot 2024-03-27 105636

## Proposal

- [x] create app features 
- [x] move files

### CODE REVIEW

**Review by lebaudantoine — APPROVED** [review]
LGTM, one comment on the overall file organization

File: src/frontend/apps/desk/src/features/app/index.ts [hunk 1]
```diff
@@ -0,0 +1,2 @@
+export * from './components/AppProvider';
+export * from './components/MainLayout';
```
[the remark this answers is not in the document]
Comment (AntoLC) [reply src/frontend/apps/desk/src/features/app/index.ts:2]: Yes, I was a bit annoyed by that as well. 
What about a `core` folder, with `auth` as well in it ?

```
├───api
├───assets
├───components
├───core
│   └───auth
│   └─AppProvider.tsx
│   └─MainLayout.tsx
├───cunningham
├───features
│   ├───addMembers
│   ├───header
│   ├───language
│   ├───members
│   ├───menu
│   └───teams
├───i18n
├───pages
├───tests
├───types
├───utils
└───__tests__
```

[the remark this answers is not in the document]
Comment (AntoLC) [reply src/frontend/apps/desk/src/features/app/index.ts:2]: `MainLayout` could be in components with a small refacto.

[the remark this answers is not in the document]
Comment (AntoLC) [reply src/frontend/apps/desk/src/features/app/index.ts:2]: The archi is more or less based on that:
https://github.com/alan2207/bulletproof-react/tree/master

[the remark this answers is not in the document]
Comment (lebaudantoine) [reply src/frontend/apps/desk/src/features/app/index.ts:2]: Alright, your suggestion with a `core` folder fits me !

**Pull request merged**